apps/web/app/api/v1/management/people/[personId]/route.ts

```diff
@@ -0,0 +1,79 @@
+import { responses } from "@/lib/api/response";
+import { getPerson } from "@formbricks/lib/services/person";
+import { DatabaseError, InvalidInputError, ResourceNotFoundError } from "@formbricks/errors";
+import { NextResponse } from "next/server";
+import { deletePerson } from "@formbricks/lib/services/person";
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { TPerson } from "@formbricks/types/v1/people";
+import { authenticateRequest } from "@/app/api/v1/auth";
+
+async function fetchAndValidatePerson(authentication: any, personId: string): Promise<TPerson | null> {
+  const person = await getPerson(personId);
+  if (!person) {
+    return null;
+  }
+  if (!(await canUserAccessPeople(authentication, person))) {
+    throw new Error("Unauthorized");
+  }
+  return person;
+}
+
+const canUserAccessPeople = async (authentication: any, person: TPerson): Promise<boolean> => {
+  if (!authentication) return false;
+
+  if (authentication.type === "session") {
+    return await hasUserEnvironmentAccess(authentication.session.user, person.environmentId);
+  } else if (authentication.type === "apiKey") {
+    return person.environmentId === authentication.environmentId;
```

Comment:
Would only use the check `person.environmentId === authentication.environmentId` in the function above